src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py

```diff
@@ -55,6 +60,78 @@ def __repr__(self):
 #                sql db                       #
 ###############################################
 
+
+class Engine(Enum):  # pylint: disable=too-few-public-methods
+    """SQL RDBMS engine type."""
+    db = 'db'
+    dw = 'dw'
+
+
+def _configure_db_create_params(
```

Comment:
Looks fine to me. It's not like the parameter combinations are defined at runtime (technically everything in Python is runtime, but you known what I mean.) This is a generic function helps to reuse the common logic for adding parameters. My only complain is this scarifies the readability for re-usability. But it is service team's decision to make.